scripts/get-sources

```diff
@@ -43,104 +50,125 @@ source "$BUILDER_DIR/scripts/common"
-    echo "--> $COMPONENT has NO_CHECK enabled"
-    echo "--> NOT Verifying tags..."
+if elementIn "$COMPONENT" $INSECURE_SKIP_CHECKING; then
     verify=false
+elif elementIn "$COMPONENT" $LESS_SECURE_SIGNED_COMMITS_SUFFICIENT; then
+    CHECK=signed-tag-or-commit
 fi
 
-if [ "$verify" == "true" ]; then
-    echo "--> Verifying tags..."
-    KEYRING_DIR_GIT="$BUILDER_DIR/keyrings/git/$(basename "$REPO")"
-    export KEYRING_DIR_GIT
-    if ! "$(dirname "$0")/verify-git-tag" $REPO $VERIFY_REF; then
+VERIFY_REF=$(git -C "$REPO" rev-parse -q --verify "$VERIFY_REF") || exit
+
+if [ "$verify" = 'false' ]; then
+    echo -e '\033[1;31m--> NOT verifying tags\033[0;0m'
+    exit 1
+else
+    if [[ $CHECK = 'signed-tag-or-commit' ]]; then
+        echo "--> Verifying tags or commits..."
+    else
+        echo "--> Verifying tags..."
+    fi
+    if ! "$BUILDER_DIR/scripts/verify-git-tag" "$REPO" "$VERIFY_REF"; then
         # if verfication failed, remove fetched content to make sure we'll not
         # use it
-        if [ "$fresh_clone" -eq 1 ]; then
+        if "$fresh_clone"; then
             rm -rf "$REPO"
         else
-            rm -f "$REPO"/.git/FETCH_HEAD
+            rm -f "$REPO/.git/FETCH_HEAD"
         fi
         exit 1
     fi
 fi
 
-if [ "$FETCH_ONLY" == "1" ]; then
+if [ "${FETCH_ONLY-}" == "1" ]; then
     exit 0
 fi
 
-CURRENT_BRANCH="$(cd $REPO; git branch | sed -n -e 's/^\* \(.*\)/\1/p')"
-if [ "$CURRENT_BRANCH" != "$BRANCH" ] || [ "$VERIFY_REF" == "HEAD" ]; then
-    pushd $REPO &> /dev/null
-    if [ -z "$NO_COLOR" ]; then
-        red="$(echo -e '\033[1;31m')"
-        green="$(echo -e '\033[1;32m')"
-        normal="$(echo -e '\033[0;0m')"
+pushd "$REPO" &> /dev/null || exit 1
+CURRENT_BRANCH="$(exec git branch | sed -n -e 's/^\* \(.*\)/\1/p'; )"
+if [ "$CURRENT_BRANCH" != "$BRANCH" ] || [ "$VERIFY_REF" = "HEAD" ]; then
+    if [ -z "${NO_COLOR=}" ]; then
+        red=$'\033[1;31m' green=$'\033[1;32m' normal=$'\033[0;0m'
+    else
+        red= green= normal=
     fi
     if [ -n "$(git name-rev --name-only $BRANCH 2> /dev/null)" ]; then
         echo "--> Switching branch from $CURRENT_BRANCH branch to ${green}$BRANCH${normal}"
-        git checkout $BRANCH || exit 1
+        git branch -f "$BRANCH" "$VERIFY_REF" &&
+        git checkout "$BRANCH" || exit 1
     else
         echo -e "--> Switching branch from $CURRENT_BRANCH branch to new ${red}$BRANCH${normal}"
-        git checkout FETCH_HEAD -b $BRANCH || exit 1
+        git checkout "$VERIFY_REF" -b $BRANCH || exit 1
     fi
-    popd &> /dev/null
 fi
 
-if [ "$VERIFY_REF" == "FETCH_HEAD" ]; then
+if ! "$fresh_clone"; then
     echo "--> Merging..."
-    pushd $REPO &> /dev/null
-    git merge --commit -q FETCH_HEAD
-    popd &> /dev/null
+    git -c merge.verifySignatures=no merge --ff-only --commit -q "$VERIFY_REF"
+    if remote=$(exec git rev-parse --symbolic-full-name '@{u}' "$VERIFY_REF" 2>/dev/null) &&
```

Comment:
Are you sure this is correct? What happens if you have a branch set to track `your-own-account/master`, but you fetch with `make get-sources` from @QubesOS (lets name matching remote `origin` here)? Wouldn't this update `refs/remote/your-own-account/master` instead of `refs/remote/origin/master`?

Reply:
Indeed it would.  I will `if false` this code out until I can figure out how to implement it correctly.